sample_formatted_code.do_not_merge.c

```diff
@@ -0,0 +1,562 @@
+    switch (a) {
+        case 13: {
+            int a = bla();
+            break;
+        }
+        case 15:
+            blu();
+            break;
+        default:
+            gugus();
+    }
+}
+
+//--- goto ---
+// https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Coding_Style#goto
+void goto_style()
+{
+    DetectFileextData *fileext = NULL;
+
+    fileext = SCMalloc(sizeof(DetectFileextData));
+    if (unlikely(fileext == NULL))
+        goto error;
+
+    memset(fileext, 0x00, sizeof(DetectFileextData));
+
+    if (DetectContentDataParse("fileext", str, &fileext->ext, &fileext->len,
+                &fileext->flags) == -1) {
+        goto error;
+    }
+
+    return fileext;
+
+error:
+    if (fileext != NULL)
+        DetectFileextFree(fileext);
+    return NULL;
+}
+
+int goto_style_nested()
+{
+    // IndentGotoLabels
+    if (foo()) {
+    label1:
+        bar();
+    }
+
+label2:
+    return 1;
+}
+
+//--- ternary style ---
+// BreakBeforeTernaryOperators - whether ? and : are on next line
+void ternary_op_style()
+{
+    // fits on one line
+    float droppy = a > 0 ? a * 100 : 0;
+
+    // split across lines - Continuation indent based on '=' fits line
+    float drop_percent = likely(ptv->last_stats64.ps_recv > 0)
+                                 ? (((float)ptv->last_stats64.ps_drop) /
+                                           (float)ptv->last_stats64.ps_recv) *
+                                           100
+                                 : 0;
+
+    // split across lines - Continuation indent based on '=' would be too long,
+    // uses normal continuation indent based on start of line
+    float drop_percent_a_bit_longer =
+            likely(ptv->last_stats64.ps_recv > 0)
+                    ? (((float)ptv->last_stats64.ps_drop) /
+                              (float)ptv->last_stats64.ps_recv) *
+                              100
+                    : 0;
+}
+
+//--- enum style ---
+// clang 11: AllowShortEnumsOnASingleLine
+// clang < 11:
+// - merges short enums on one line if BraceWrapping: AfterEnum: false
+// - one-value-by-line if BraceWrapping: AfterEnum: true
+enum Gugus { bla, bli, blu };
+
+enum ThisIsTooLongForOneLine {
+    blablablablablablablabla,
+    blibliblibliblibliblibli,
+    blublublublublublublublu
+};
+
+enum { A, B } myEnum;
+
+// trailing comma forces one-value-by-line
+enum {
+    NFS_DECODER_EVENT_EMPTY_MESSAGE,
+};
+
+//--- union style ---
+typedef union {
+    int gugus;
+} Bla;
+
+union bla {
+    int gugus;
+};
+
+// --- struct style ---
+struct bla {
+    int gugus;
+};
+
+struct bla_ {
+    int gugus;
+} Bla;
+
+typedef struct bla_ {
+    int gugus;
+} Bla;
+
+//--- Alignment ---
+struct bla {
+    // AlignConsecutiveDeclarations
+    // AlignTrailingComments
+    int a;       /* comment */
+    unsigned bb; /* comment */
+    int *ccc;    /* comment */
+};
+
+// pointers
+// DerivePointerAlignment
+// PointerAlignment
+void *ptr;
```

Comment:
I saw mostly right-aligned pointers so that's what I set up.

DerivePointerAlignment could be useful for whole-file formatting to "follow the prevalent left/centre/right aligned style". I disabled that.

Reply:
yeah right aligned it is